In [1]:
from pyproj import Transformer

#### Hypocentres

In [2]:
#transformer = Transformer.from_crs("epsg:32627", "epsg:4326", always_xy=True)
transformer = Transformer.from_crs("utm27", "epsg:4326", always_xy=True)

|                	| x           	| y            	| z     	|
|----------------	|-------------	|--------------	|-------	|
| Simple East    	| 670950.2    	| 7327297.5    	| -7000 	|
| Simple Middle  	| 639613.9    	| 7340258.5    	| -7000 	|
| Simple West    	| 606405.1    	| 7357136.2    	| -7000 	|
| Complex East   	| 663105.107  	| 7331567.26   	| -7000 	|
| Complex Middle 	| 642704.0    	| 7339196.2    	| -7000 	|
| Complex West   	| 607077.3925 	| 7349647.4498 	| -7000 	|

In [11]:
x = [670950.2, 639613.9, 606405.1, 663105.107, 642704.0, 607077.3925]
y = [7327297.5, 7340258.5, 7357136.2, 7331567.26, 7339196.2, 7349647.4498]

In [12]:
# project back to geocentric coordinates
lon, lat = transformer.transform(x, y)

In [13]:
print(lon)

[-17.228592182059785, -17.904368452344116, -18.625532543980157, -17.395811868445602, -17.837175366287514, -18.616907927623227]


In [14]:
print(lat)

[66.01825068509896, 66.14971339178402, 66.31396741623851, 66.06061423001546, 66.13881332034997, 66.24661319458696]


|                	| lon      	| lat     	|
|----------------	|----------	|---------	|
| Simple East    	| -17.2286 	| 66.0201 	|
| Simple Middle  	| -17.9044 	| 66.1515 	|
| Simple West    	| -18.6257 	| 66.3158 	|
| Complex East   	| -17.3958 	| 66.0625 	|
| Complex Middle 	| -17.8373 	| 66.1406 	|
| Complex West   	| -18.6170 	| 66.2485 	|

#### Modal domain

In [15]:
x_dom = [525000, 730000]
y_dom = [7.248e+06, 7.43e+06]

In [16]:
lon_dom, lat_dom = transformer.transform(x_dom, y_dom)
print(lon_dom, lat_dom)

[-20.462392387118324, -15.741480005402638] [65.35199350163427, 66.89855672600817]


## POI 

x 	y 	time 	h 	b 	waterlevel 	ssha

    Sig
**596010 	7342010 	0.0 	22.827670 	-22.827670 	0.000000 	0.000000**  

    Olaf  
**608100 	7332300 	0.0 	16.483257 	-16.483257 	0.000000 	0.000000**     

    Dalvik  
**613300 	7318400 	0.0 	5.180788 	-5.180788 	0.000000 	0.000000**  

    Akureyri  
**632000 	7292100 	0.0 	37.141219 	-37.141219 	0.000000 	0.000000**   

    Grimsey 
**633500 	7381000 	0.0 	10.304572 	-10.304572 	0.000000 	0.000000**  

    Husavik  
**664500 	7329200 	0.0 	7.298772 	-7.298772 	0.000000 	0.000000**  

In [17]:
x_POI = [596010, 608100, 613300, 632000, 633500, 664500]
y_POI = [7342010, 7332300, 7318400,  7292100, 7381000, 7329200]

lon_POI, lat_POI = transformer.transform(x_POI, y_POI)
print(lon_POI, lat_POI)

[-18.86874200400823, -18.60890580976148, -18.506254240201283, -18.121833089188257, -17.99634741706946, -17.368093913939497] [66.1817270388725, 66.0907693517237, 65.96436250066692, 65.72149180883459, 66.51736523783475, 66.03869709441048]


In [18]:
transformer = Transformer.from_crs("epsg:32627", "epsg:4326", always_xy=True)
lon_POI, lat_POI = transformer.transform(x_POI, y_POI)
print(lon_POI, lat_POI)

[-18.868954963838295, -18.60909621777071, -18.506435260672824, -18.121980855704454, -17.99648792380457, -17.368179678603134] [66.18361293164726, 66.09265463740168, 65.96624664147667, 65.72337387403358, 66.51925524380808, 66.04058266307385]


#### Box around the fault based on Lat Long

In [8]:
# fault - change transformer
transformer = Transformer.from_crs("epsg:4326", "utm27", always_xy=True)
x_f = [-18.9, -16.9]
y_f=[65.9, 66.35]
x_f1, y_f1 = transformer.transform(x_f, y_f)

In [9]:
print(x_f1, y_f1)

[595653.8326967588, 683408.9984408879] [7310573.975060897, 7365147.965061599]


#### Reproject coordinates of HFFZ simple fault geometry connected

In [3]:
import pandas as pd

In [23]:
df = pd.read_csv('/import/freenas-m-05-seissol/kutschera/HIWI/Scripts/Github/GMT/Iceland/input/HFF_fault_traces_simple_connected/fault_1.dat',
                names=['x','y','z'], sep='\s+')
df.head()

x             y    z
0  603915.1970  7.358621e+06  0.0
1  604106.7255  7.358505e+06  0.0
2  604298.2540  7.358390e+06  0.0
3  604489.7825  7.358277e+06  0.0
4  604681.3111  7.358164e+06  0.0

In [24]:
df.dtypes

x    float64
y    float64
z    float64
dtype: object

In [26]:
transformer = Transformer.from_crs("utm27", "epsg:4326", always_xy=True)
# project back to geocentric coordinates
lon, lat = transformer.transform(df.x, df.y)

In [27]:
lon = pd.Series(lon)
lat = pd.Series(lat)
  
frame = { 'Lon': lon, 'Lat': lat }
  
coord = pd.DataFrame(frame)
coord.head()

Lon        Lat
0 -18.679802  66.328112
1 -18.675627  66.327011
2 -18.671451  66.325921
3 -18.667274  66.324838
4 -18.663097  66.323762

In [29]:
coord.to_csv("/import/freenas-m-05-seissol/kutschera/HIWI/Scripts/Github/GMT/Iceland/input/HFF_fault_traces_simple_connected/fault_1_lon_lat.dat", 
             index=False, header=False, sep=' ')